checkout-core/src/main/java/com/adyen/checkout/core/model/ModelUtils.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 17/12/2020.
+ */
+package com.adyen.checkout.core.model
+
+import com.adyen.checkout.core.exception.BadModelException
+import com.adyen.checkout.core.exception.NoConstructorException
+import org.json.JSONArray
+import org.json.JSONObject
+import java.lang.reflect.Modifier
+import java.util.*
+
+class ModelUtils private constructor() {
+    companion object {
+        const val SERIALIZER_FIELD_NAME = "SERIALIZER"
+
+        /**
+         * Parse a [JSONObject] to a class that extends [ModelObject] using its [ModelObject.Serializer].
+         *
+         * @param jsonObject The object to be parsed.
+         * @param modelClass The class type to be parsed to.
+         * @param <T> The type o the ModelObject class to be parse to.
+         * @return The parsed object.
+         */
+        @JvmStatic
+        fun <T : ModelObject?> deserializeModel(jsonObject: JSONObject, modelClass: Class<T>): T {
```

Comment:
Is the nullability intended in all these methods?

Reply:
Nope 👍